emails/models.py

```diff
@@ -1,17 +1,22 @@
-from datetime import datetime
+from collections import namedtuple
+from datetime import datetime, timedelta, timezone
 from hashlib import sha256
 import random
 import string
 import uuid
 
 from django.apps import apps
+from django.conf import settings
 from django.contrib.auth.models import User
 from django.db import models
 
 
 emails_config = apps.get_app_config('emails')
 
 
+BounceStatus = namedtuple('BounceStatus', 'paused type')
```

Comment:
It's merged now, but for future ref you can leverage the named tuple more to make the code more readable and less error prone (always used keyword args to help reduce errors)

```python
BounceStatus = namedtuple('active', 'type')

def check_bounce_pause(self):
    .....
    # example return 
    return BounceStatus(active=True, type='hard')
 
def next_email_try(self):
    ...
    # example consumption
    bounce_status = self.check_bounce_pause()
    if not bounce_status.active:
         ...
    if bounce_status.type == 'soft':
        return ...
```

Reply:
But that doesn't fit in a regex replace statement in vim. 😛